src/proxy/outbound.rs

```diff
@@ -477,7 +477,7 @@ impl OutboundConnection {
         let source_workload = match self
             .pi
             .state
-            .wait_for_workload(&downstream_network_addr, Duration::from_millis(500))
+            .wait_for_workload(&downstream_network_addr, Duration::from_secs(5))
```

Comment:
Yeah fair enough. It scares me to wait this long but we gotta wait, one way or the other.